src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/WorkspaceProjectStateChangeDetector.cs

```diff
@@ -90,6 +90,13 @@ internal virtual void Workspace_WorkspaceChanged(object sender, WorkspaceChangeE
                         if (TryGetProjectSnapshot(project?.FilePath, out var _))
                         {
                             EnqueueUpdate(e.ProjectId);
+
+                            var dependencyGraph = e.NewSolution.GetProjectDependencyGraph();
+                            var dependentProjectIds = dependencyGraph.GetProjectsThatTransitivelyDependOnThisProject(e.ProjectId);
```

Comment:
Does this do dependencies-of-dependencies?
Does this include projects that directly depend on this project (rather than _exclusively_ transitive?)

I suspect tests will answer these questions, so feel free to ignore them until then

Reply:
It definitely does direct dependencies since that's what I tested on, and though I haven't actually tested a transitive dependency yet the name is terrible if that doesn't work. Directly testing this part will be hard outside of full DDRIT tests. I imagined writing unit tests and testing by hand, then following up with https://github.com/dotnet/aspnetcore/issues/26977#issuecomment-886840161.